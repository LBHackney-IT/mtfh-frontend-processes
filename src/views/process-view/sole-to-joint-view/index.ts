export * from "./sole-to-joint-view";
export * from "./states";
