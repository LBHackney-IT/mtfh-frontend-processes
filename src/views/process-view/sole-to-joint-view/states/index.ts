export * from "./select-tenants-view";
export * from "./check-eligibility-view";
export * from "./submit-case-view";
export * from "./review-documents-view";
export * from "./request-documents-view";
export * from "./breach-checks-view";
export * from "./tenure-investigation-view";
export * from "./close-process-view";
