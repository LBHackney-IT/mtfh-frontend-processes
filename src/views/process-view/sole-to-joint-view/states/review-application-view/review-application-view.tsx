import React, { useState } from "react";

import { CloseProcessView } from "../../../../../components";
import { locale } from "../../../../../services";
import { IProcess } from "../../../../../types";
import { isCurrentState } from "../../../../../utils/processUtil";
import { DesBox } from "../../../shared/process-components";
import { SubmitCaseView } from "../../../shared/submit-case-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import {
  EligibilityChecksPassedBox,
  TenureInvestigationRecommendationBox,
} from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
} from "@mtfh/common/lib/components";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const [documentsSigned, setDocumentsSigned] = useState<boolean>(false);
  const { submitted, closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  if (applicationSubmitted.state === process.currentState.state && submitted) {
    return (
      <SubmitCaseView
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        optional={optional}
      />
    );
  }

  return (
    <div data-testid="soletojoint-ReviewApplication">
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox
        title={views.submitCase.supportingDocumentsApproved}
        description={views.submitCase.viewDocumentsOnDes}
      />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <TenureInvestigationRecommendationBox
          processConfig={processConfig}
          process={process}
        />
      )}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
        tenureUpdated.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{
            closeCase,
            setCloseCase,
            documentsSigned,
            setDocumentsSigned,
            tenant,
          }}
        />
      )}

      {closeCase && (
        <>
          <Box variant="warning">
            <StatusHeading
              variant="warning"
              title={views.closeProcess.soleToJointClosed}
            />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
            setGlobalError={setGlobalError}
          />
        </>
      )}
    </div>
  );
};
