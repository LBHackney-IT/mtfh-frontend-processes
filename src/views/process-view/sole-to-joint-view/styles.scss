@import "../../../../node_modules/lbh-frontend/lbh/settings/all";

.lbh-dialog {
  max-width: 510px;
  &__close {
    display: none;
  }
  &__actions {
    .lbh-button {
      width: 15em;
    }
  }
}

.divider {
  margin: 1.5em 0 1.5em 0;
  border-top: 2px solid lbh-colour("lbh-a01");
}

.comment-form {
  margin-bottom: 1.5em;
}

.closeCommentDialogActions {
  flex-direction: column;
  align-items: center;
  row-gap: 1.5em;

  .lbh-button--secondary {
    margin-left: 0;
  }
}
