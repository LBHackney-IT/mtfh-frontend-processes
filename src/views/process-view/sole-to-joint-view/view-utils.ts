import { processes } from "../../../services";

const { states } = processes.soletojoint;

const {
  documentsRequestedDes,
  documentsRequestedAppointment,
  documentsAppointmentRescheduled,
} = states;

export const reviewDocumentsStates = [
  documentsRequestedDes.state,
  documentsRequestedAppointment.state,
  documentsAppointmentRescheduled.state,
];
