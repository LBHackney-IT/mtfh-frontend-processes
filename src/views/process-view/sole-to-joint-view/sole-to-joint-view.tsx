import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { CloseProcessView, SoleToJointHeader } from "../../../components";
import { CloseCaseButton } from "../../../components/close-case-button/close-case-button";
import { locale, processes } from "../../../services";
import { ProcessComponentProps, ProcessSideBarProps } from "../../../types";
import { getPreviousState, isSameState } from "../../../utils/processUtil";
import {
  BreachChecksFailedView,
  BreachChecksView,
  CheckEligibilityView,
  RequestDocumentsView,
  ReviewDocumentsView,
  SelectTenantsView,
  SubmitCaseView,
} from "./states";
import { ManualChecksFailedView } from "./states/manual-checks-view";
import { ReviewApplicationView } from "./states/review-application-view/review-application-view";
import { reviewDocumentsStates } from "./view-utils";

import { Button, ErrorSummary, Step, Stepper } from "@mtfh/common/lib/components";
import { useFeatureToggle } from "@mtfh/common/lib/hooks";

import "./styles.scss";

const { views } = locale;
const { soleToJoint } = views;
const processConfig = processes.soletojoint;

const { states } = processConfig;
const {
  selectTenants,
  automatedChecksFailed,
  automatedChecksPassed,
  manualChecksFailed,
  manualChecksPassed,
  breachChecksPassed,
  breachChecksFailed,
  documentChecksPassed,
  applicationSubmitted,
  tenureInvestigationFailed,
  tenureInvestigationPassed,
  tenureInvestigationPassedWithInt,
  interviewScheduled,
  interviewRescheduled,
  hoApprovalPassed,
  hoApprovalFailed,
  tenureAppointmentScheduled,
  tenureAppointmentRescheduled,
  tenureUpdated,
  processCancelled,
  processClosed,
} = states;

const reviewDocumentsViewByStates = {};
reviewDocumentsStates.forEach((state) => {
  reviewDocumentsViewByStates[state] = ReviewDocumentsView;
});

const components = {
  [selectTenants.state]: SelectTenantsView,
  [automatedChecksFailed.state]: CheckEligibilityView,
  [automatedChecksPassed.state]: CheckEligibilityView,
  [manualChecksFailed.state]: ManualChecksFailedView,
  [manualChecksPassed.state]: BreachChecksView,
  [breachChecksFailed.state]: BreachChecksFailedView,
  [breachChecksPassed.state]: RequestDocumentsView,
  ...reviewDocumentsViewByStates,
  [documentChecksPassed.state]: SubmitCaseView,
  [applicationSubmitted.state]: ReviewApplicationView,
  [tenureInvestigationFailed.state]: ReviewApplicationView,
  [tenureInvestigationPassed.state]: ReviewApplicationView,
  [tenureInvestigationPassedWithInt.state]: ReviewApplicationView,
  [interviewScheduled.state]: ReviewApplicationView,
  [interviewRescheduled.state]: ReviewApplicationView,
  [hoApprovalPassed.state]: ReviewApplicationView,
  [hoApprovalFailed.state]: ReviewApplicationView,
  [tenureAppointmentScheduled.state]: ReviewApplicationView,
  [tenureAppointmentRescheduled.state]: ReviewApplicationView,
  [tenureUpdated.state]: ReviewApplicationView,
  [processCancelled.state]: SubmitCaseView,
  [processClosed.state]: CheckEligibilityView,
};

const getActiveStep = (
  process: any,
  states,
  submitted: boolean,
  closeProcessReason?: string,
) => {
  const {
    currentState: { state },
  } = process;

  if (state === states.selectTenants.state) {
    return 0;
  }

  if ([processCancelled.state, processClosed.state].includes(state)) {
    const previousState = getPreviousState(process);
    if (
      [
        manualChecksFailed.state,
        automatedChecksPassed.state,
        automatedChecksFailed.state,
      ].includes(previousState.state)
    ) {
      return 1;
    }
    if (
      [
        manualChecksPassed.state,
        breachChecksFailed.state,
        breachChecksPassed.state,
      ].includes(previousState.state)
    ) {
      return 2;
    }
    if (
      [
        hoApprovalFailed.state,
        interviewScheduled.state,
        interviewRescheduled.state,
        hoApprovalPassed.state,
        tenureInvestigationPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
      ].includes(previousState.state)
    ) {
      return 7;
    }
  }

  if (
    [
      states.automatedChecksPassed.state,
      states.automatedChecksFailed.state,
      states.manualChecksFailed.state,
    ].includes(state)
  ) {
    return 1;
  }
  if (
    [
      states.manualChecksPassed.state,
      states.breachChecksFailed.state,
      states.processCancelled.state,
    ].includes(state) ||
    (states.manualChecksPassed.state === state && submitted)
  ) {
    return 2;
  }
  if (state === states.breachChecksPassed.state) {
    return 3;
  }
  if (
    [
      states.documentsRequestedDes.state,
      states.documentsRequestedAppointment.state,
      states.documentsAppointmentRescheduled.state,
      states.processClosed.state,
    ].includes(state)
  ) {
    return 4;
  }
  if (state === states.documentChecksPassed.state) {
    return 5;
  }
  if (
    [
      states.applicationSubmitted.state,
      states.tenureInvestigationFailed.state,
      states.tenureInvestigationPassed.state,
      states.tenureInvestigationPassedWithInt.state,
      states.interviewScheduled.state,
      states.interviewRescheduled.state,
      states.hoApprovalPassed.state,
      states.tenureAppointmentScheduled.state,
      states.tenureAppointmentRescheduled.state,
    ].includes(state) ||
    (states.applicationSubmitted.state === state && submitted)
  ) {
    if (closeProcessReason) {
      return 7;
    }
    return 6;
  }
  if ([states.tenureUpdated.state, states.hoApprovalFailed.state].includes(state)) {
    return 7;
  }
  return 0;
};

export const SoleToJointSideBar = (props: ProcessSideBarProps) => {
  const hasReassignCase = useFeatureToggle("MMH.ReassignCase");
  const {
    process,
    process: {
      id: processId,
      processName,
      currentState: { state },
    },
    submitted = false,
    closeProcessReason,
    setCloseProcessDialogOpen,
    setCancel,
  } = props;

  let activeStep = getActiveStep(process, states, submitted, closeProcessReason);
  let steps: JSX.Element[];
  let startIndex = 0;
  if (activeStep > 6 || (!submitted && activeStep === 6)) {
    steps = [
      <Step key="step-review-application">{soleToJoint.steps.reviewApplication}</Step>,
      <Step key="step-end-case">{soleToJoint.steps.endCase}</Step>,
    ];
    activeStep = activeStep - 6;
    startIndex = 10;
  } else if (
    activeStep > 2 ||
    (!submitted && processCancelled.state !== state && activeStep === 2)
  ) {
    steps = [
      <Step key="step-breach-of-tenancy">{soleToJoint.steps.breachOfTenancy}</Step>,
      <Step key="step-request-documents">{soleToJoint.steps.requestDocuments}</Step>,
      <Step key="step-review-documents">{soleToJoint.steps.reviewDocuments}</Step>,
      <Step key="step-submit-case">{soleToJoint.steps.submitCase}</Step>,
      <Step key="step-finish">{soleToJoint.steps.finish}</Step>,
    ];
    activeStep = activeStep - 2;
    startIndex = 3;
  } else {
    steps = [
      <Step key="step-select-tenant">{soleToJoint.steps.selectTenant}</Step>,
      <Step key="step-personal-details">{soleToJoint.steps.checkEligibility}</Step>,
      <Step key="step-finish">{soleToJoint.steps.finish}</Step>,
    ];
  }
  return (
    <>
      <Stepper
        data-testid="mtfh-stepper-sole-to-joint"
        activeStep={activeStep}
        startIndex={startIndex}
      >
        {steps}
      </Stepper>
      {hasReassignCase && (
        <Button variant="secondary">{soleToJoint.actions.reassignCase}</Button>
      )}
      {[
        automatedChecksPassed.state,
        manualChecksPassed.state,
        breachChecksPassed.state,
        tenureInvestigationPassed.state,
        interviewScheduled.state,
        interviewRescheduled.state,
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
      ].includes(state) &&
        !closeProcessReason && (
          <Button
            variant="secondary"
            onClick={() => {
              setCancel(true);
              setCloseProcessDialogOpen(true);
            }}
          >
            {soleToJoint.actions.cancelProcess}
          </Button>
        )}
      <Button
        variant="secondary"
        as={RouterLink}
        to={`/activities/process/${processName}/${processId}`}
      >
        {soleToJoint.actions.caseActivityHistory}
      </Button>
    </>
  );
};

const getComponent = (process) => {
  const {
    currentState: { state },
  } = process;

  if ([processCancelled.state, processClosed.state].includes(state)) {
    const previousState = getPreviousState(process);
    if (isSameState(previousState, manualChecksFailed)) {
      return ManualChecksFailedView;
    }
    if (isSameState(previousState, breachChecksFailed)) {
      return BreachChecksFailedView;
    }
    return components[previousState.state];
  }
  return components[state];
};

export const SoleToJointView = ({ process, mutate, optional }: ProcessComponentProps) => {
  const {
    closeProcessReason,
    submitted,
    setSubmitted,
    setCloseProcessDialogOpen,
    isCancel,
  } = optional;

  const Component = getComponent(process);

  if (!Component) {
    return (
      <ErrorSummary
        id="sole-to-joint-view"
        data-testid="error-sole-to-joint-view"
        title={locale.errors.unableToFindState}
        description={locale.errors.unableToFindStateDescription}
      />
    );
  }

  return (
    <>
      <SoleToJointHeader processConfig={processConfig} process={process} />
      <Component
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        optional={{
          submitted,
          setSubmitted,
          closeProcessReason,
          setCloseProcessDialogOpen,
        }}
      />

      {(closeProcessReason ||
        [
          processClosed.state,
          processCancelled.state,
          automatedChecksFailed.state,
          breachChecksFailed.state,
          manualChecksFailed.state,
          hoApprovalFailed.state,
        ].includes(process.currentState.state)) && (
        <CloseProcessView
          closeProcessReason={closeProcessReason}
          process={process}
          processConfig={processConfig}
          mutate={mutate}
          isCancel={isCancel}
        />
      )}

      {!closeProcessReason &&
        reviewDocumentsStates.includes(process.currentState.state) && (
          <CloseCaseButton setCloseProcessDialogOpen={setCloseProcessDialogOpen} />
        )}
    </>
  );
};
