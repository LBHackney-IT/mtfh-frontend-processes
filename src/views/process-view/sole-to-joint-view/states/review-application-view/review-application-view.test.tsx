import {
  getContactDetailsV2,
  getReferenceDataV1,
  getTenureV1,
  mockContactDetailsV2,
  mockProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";

import { locale, processes } from "../../../../../services";
import { SoleToJointView } from "../../sole-to-joint-view";
import { ReviewApplicationView } from "./review-application-view";

let submitted = false;
let closeCase = false;
const setSubmitted = () => {};
const setCloseCase = () => {};

const options = {
  url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

describe("tenure-investigation-view", () => {
  beforeEach(() => {
    jest.resetModules();
    submitted = false;
    closeCase = false;
  });

  test("it renders ReviewApplication view correctly for ApplicationSubmitted state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.tenureInvestigationCompleted, {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders no tenant found if no tenant", async () => {
    server.use(getTenureV1({ householdMembers: [] }));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await expect(
      screen.findByText("Tenant not found.", {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders error if tenure cannot be fetched", async () => {
    server.use(getTenureV1(null, 500));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.errors.unableToFetchRecordDescription, {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewApplication view correctly for close case", async () => {
    closeCase = true;
    server.use(getContactDetailsV2(mockContactDetailsV2));
    render(
      <SoleToJointView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureAppointmentRescheduled",
            processData: {
              formData: {
                appointmentDateTime: "2010-10-17T08:59:00.000Z",
              },
              documents: [],
            },
          },
          previousStates: [
            {
              ...mockProcessV1.currentState,
              state: "TenureAppointmentScheduled",
              processData: {
                formData: {
                  appointmentDateTime: "2010-10-12T08:59:00.000Z",
                },
                documents: [],
              },
            },
          ],
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.soleToJointWillBeClosed),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewApplication view correctly for TenureUpdated state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(getReferenceDataV1());
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          processName: "soletojoint",
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureUpdated",
          },
          relatedEntities: [
            {
              id: "f8daaf21-edfc-4abc-9c39-403b84a142cb",
              targetType: "tenure",
              subType: "newTenure",
              description: "New Tenure created for this process.",
            },
          ],
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.views.tenureInvestigation.tenancySigned),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.tenureInvestigation.viewNewTenure),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.closeProcess.thankYouForConfirmation),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
