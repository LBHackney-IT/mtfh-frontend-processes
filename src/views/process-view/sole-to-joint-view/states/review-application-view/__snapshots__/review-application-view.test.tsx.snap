// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tenure-investigation-view it renders ReviewApplication view correctly for ApplicationSubmitted state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
          target="_blank"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <div
      class="govuk-checkboxes__item"
    >
      <input
        class="govuk-checkboxes__input"
        id="tenure-investigation-completed"
        type="checkbox"
      />
      <label
        class="govuk-label govuk-checkboxes__label"
        for="tenure-investigation-completed"
      >
        I confirm that the tenure investigation has been completed
      </label>
    </div>
    <p
      class="lbh-body-m"
    >
      Recommendation:
    </p>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Approve
    </button>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Appointment
    </button>
    <button
      class="govuk-button lbh-button"
      disabled=""
      style="width: 222px; margin-right: 100%;"
      type="submit"
    >
      Decline
    </button>
  </div>
</div>
`;

exports[`tenure-investigation-view it renders ReviewApplication view correctly for TenureUpdated state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
          target="_blank"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Tenure investigator recommendation: Interview Applicant
        </h4>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Sole to joint tenure application approved
        </h4>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Tenancy signed: tenure created
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/tenure/f8daaf21-edfc-4abc-9c39-403b84a142cb"
        >
          View new tenure
        </a>
      </div>
    </div>
    <h3
      class="lbh-heading-h3"
    >
      Thank you for your confirmation
    </h3>
    <ul
      class="lbh-list lbh-list--bullet"
      style="margin-left: 1em;"
    >
      <li>
        <p
          class="lbh-body-s"
        >
          This case is now closed and we have record this on the system - that you have sent an outcome letter to the resident. The outcome can be viewed in the activity history
        </p>
      </li>
    </ul>
    <div
      style="margin-top: 1em;"
    >
      <a
        class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
        href="/"
      >
        Return to home page
      </a>
    </div>
  </div>
</div>
`;
